controller_manager:
  ros__parameters:
    update_rate: 100
    use_sim_time: true

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster
      # use_local_topics: true  # keep commented unless you know you need it

    joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController

joint_state_broadcaster:
  ros__parameters:
    use_sim_time: true

joint_trajectory_controller:
  ros__parameters:
    joints:
      - joint2_to_joint1
      - joint3_to_joint2
      - joint4_to_joint3
      - joint5_to_joint4
      - joint6_to_joint5
      - joint6output_to_joint6
    command_interfaces: [position]
    state_interfaces: [position, velocity]
    state_publish_rate: 50.0
    action_monitor_rate: 20.0
    allow_partial_joints_goal: false
    constraints:
      goal_time: 2.0
      stopped_velocity_tolerance: 0.01
      joint2_to_joint1: { trajectory: 1.0, goal: 0.01 }
      joint3_to_joint2: { trajectory: 1.0, goal: 0.01 }
      joint4_to_joint3: { trajectory: 1.0, goal: 0.01 }
      joint5_to_joint4: { trajectory: 1.0, goal: 0.01 }
      joint6_to_joint5: { trajectory: 1.0, goal: 0.01 }
      joint6output_to_joint6: { trajectory: 1.0, goal: 0.01 }

